Adding item.disabled property to disable checkboxes

MiguelTolosa opened the pull request:
Fixes #

## Proposed Changes

I would like to add the ability to set checkboxes as disabled through an optional 'disabled' property in the Item.item.  In this way the passed List of 'items' from the user can be formatted as 

items: [
        { id: 0, label: "item 1", disabled: false },
        { id: 2, label: "item 2", disabled: true },
        { id: 3, label: "item 3", disabled: false },
        { id: 4, label: "item 4" },
        { id: 5, label: "item 5" }
      ],

This helps the user manage the checkboxes corresponding to their items List directly in the structure without having to do additional work aside from set a new propery.  This is helpful for React apps that do a lot of back end calls since the 'disabled' property can dynamically be added to the List item if an item needs to be disabled.

This is added as a new addition of (item.disabled === true), which will only set the checkbox to 'disabled' if the item was passed a disabled property and the passed property is true.
  -
  -
  -

liorheber commented:
@MiguelTolosa thank you for contributing, it's a great idea!
I have a few minor comments:
1. There's also a disabled style in the default renderer, it's worth using it when the item is disabled.
2. Could you add a test to the scenario for an item that receives a disabled props to ensure that we do not accidentally break your change in the future.
3. Could you add a disabled item to the example in the readme file so that the change is documented?

Thanks again!

MiguelTolosa commented:
Sounds good, I have added tests and edited the readme to reflect the changes.  

There isn't documentation for how to use the default renderer and its styles, which is why I implemented this version for development until I could figure that one out.

liorheber commented on src/components/items/item.js:
```diff
@@ -29,7 +29,7 @@ const Item = ({
       checked={checked}
       indeterminate={indeterminate}
-      disabled={disabled}
+      disabled={disabled || item.disabled === true}
```
@MiguelTolosa sorry last few comments I hope!
This line seems like a duplication of the same logic at list.js@74
You're going to pass a disabled=true from the list so I'm not sure there's a reason to check it again.

liorheber replied:
yup, that's the last comment :)

stepovat commented:
Hi @MiguelTolosa, 
accidentally saw your proposed changes and want to tell that this was implemented in 1.0.58 version of component... PR with changes https://github.com/kenshoo/react-multi-select/pull/101
Best regards!
